crates/lunatic-process-api/src/lib.rs

```diff
@@ -665,21 +673,19 @@ fn unlink<T: ProcessState + ProcessCtx<T>>(
     let this_process = WasmProcess::new(id, signal_mailbox.0);
 
     // Send unlink signal to other process
-    let process = caller
-        .data()
-        .environment()
-        .get_process(process_id)
-        .or_trap("lunatic::process::link")?
-        .clone();
-    process.send(Signal::UnLink(Arc::new(this_process)));
-
-    // Send unlink signal to itself
-    caller
-        .data_mut()
-        .signal_mailbox()
-        .0
-        .send(Signal::UnLink(process))
-        .expect("The signal is sent to itself and the receiver must exist at this point");
+    let process = caller.data().environment().get_process(process_id);
+
+    if let Some(process) = process {
+        process.send(Signal::UnLink(Arc::new(this_process)));
+
+        // Send unlink signal to itself
+        caller
+            .data_mut()
+            .signal_mailbox()
+            .0
+            .send(Signal::UnLink(process))
+            .expect("The signal is sent to itself and the receiver must exist at this point");
```

Comment:
The unlink signal should always be sent to itself, even if the other process doesn't exist. Keeping the link locally doesn't do anything and just consumes resources.

Reply:
OK, but that was the problem as `Signal::UnLink` containts `Process` which in this case doesn't exist.

Reply:
I will change `UnLink` to have process id, not `Process`. It's simple.